posthog/queries/event_query/event_query.py

```diff
@@ -113,14 +115,15 @@ def _determine_should_join_persons(self) -> None:
             self._should_join_persons = True
             return
 
-        if any(
-            self._should_property_join_persons(prop)
-            for entity in self._filter.entities
-            for prop in entity.property_groups.flat
-        ):
-            self._should_join_distinct_ids = True
-            self._should_join_persons = True
-            return
+        for entity in self._filter.entities:
+            if (
+                entity.math in MATH_FUNCTIONS
```

Comment:
it doesn't make sense for this to be in the general base event query, since this is specific to trends only.

I'd recommend adding this here instead: https://github.com/PostHog/posthog/blob/master/posthog/queries/trends/trend_event_query.py#L115

Reply:
Fair point, moved